pkg/machinepool/machinepool.go

```diff
@@ -1442,19 +1442,21 @@ func getMachinePoolReplicas(cmd *cobra.Command,
 				return
 			}
 		}
-	} else if interactive.Enabled() || !isReplicasSet && askForScalingParams {
+	} else {
 		if !isReplicasSet {
 			replicas = existingReplicas
 		}
-		replicas, err = interactive.GetInt(interactive.Input{
-			Question: "Replicas",
-			Help:     cmd.Flags().Lookup("replicas").Usage,
-			Default:  replicas,
-			Required: true,
-		})
-		if err != nil {
-			err = fmt.Errorf("Expected a valid number of replicas: %s", err)
-			return
+		if interactive.Enabled() || !isReplicasSet && askForScalingParams {
+			replicas, err = interactive.GetInt(interactive.Input{
+				Question: "Replicas",
+				Help:     cmd.Flags().Lookup("replicas").Usage,
+				Default:  replicas,
+				Required: true,
+			})
+			if err != nil {
+				err = fmt.Errorf("Expected a valid number of replicas: %s", err)
+				return
```

Comment:
We use naked return extensively. Naked returns should be used only for very simple functions since they make it hard to understand what we are returning.

Extract from the `go` website:
```
Naked return statements should be used only in short functions, as with the example shown here. They can harm readability in longer functions.
```
I suggest opening another PR to refactor this function or creating an issue to keep track of this.

Reply:
@ziccardi 
thanks for the comments
I will create an issue to track code refactoring
- split the large method
- don't use naked return
for this pr, let it focus on fixing the blocker issue